packages/rx_bloc_cli/mason_templates/bricks/feature_profile/__brick__/lib/feature_profile/blocs/profile_bloc.dart

```diff
@@ -46,37 +45,41 @@ class ProfileBloc extends $ProfileBloc {
   Stream<bool> _mapToIsLoadingState() => loadingState;
 
   @override
-  ConnectableStream<Result<bool>> _mapToSyncNotificationsStatusState() =>
+  ConnectableStream<Result<bool>> _mapToAreNotificationsEnabledState() =>
       Rx.merge([
-        _$setNotificationsEvent.switchMap(
+        _$toggleNotificationsEvent
+            .switchMap((_) =>
+                _notificationService.areNotificationsEnabled().asStream())
+            .switchMap(
+              (areEnabled) => !areEnabled
+                  ? _notificationService
```

Comment:
Do you think combining the check for enabled mortifications and requesting them in a single switch  map will be a better option?

Reply:
Moved to service